include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp

```diff
@@ -0,0 +1,417 @@
+            auto shifted_coord = coord;
+
+            shifted_coord.x += 1;
+            shifted_coord.y += 2;
+
+            const auto nm_pos = sidb_nm_position(lyt, shifted_coord);
+
+            svg_content << generate_lattice_point(nm_pos.first * 10, nm_pos.second * 10, fiction::svg::SI_LATTICE);
+        }
+
+        std::vector<cell<Lyt>> all_cells{};
+        all_cells.reserve(lyt.num_cells());
+        // collect all cells
+        lyt.foreach_cell([&all_cells](const auto& cell) { all_cells.push_back(cell); });
+        std::sort(all_cells.begin(), all_cells.end());
+
+        for (const auto& cell : all_cells)
+        {
+            // Shift coordinates for alignment
+            auto shifted_cell = cell;
+
+            // shift for padding
+            shifted_cell.x += 1;
+            shifted_cell.y += 2;
+
+            const auto nm_pos = sidb_nm_position(lyt, shifted_cell);
+
+            if constexpr (is_charge_distribution_surface_v<Lyt>)
+            {
+                // If the layout has charge distribution information
+                const auto charge_state = lyt.get_charge_state(cell);
+                svg_content << generate_sidb(nm_pos.first * 10, nm_pos.second * 10, charge_state);
+            }
+            else
+            {
+                // Default SiDB dot without charge state
+                svg_content << generate_sidb(nm_pos.first * 10, nm_pos.second * 10);
+            }
+        };
+
+        auto shifted_max = max_coord;
+
+        // shift for padding
+
+        shifted_max.x += 2;
+        shifted_max.y += 3;
+
+        // Compute viewBox dimensions
+        const auto view_box_nm_min = sidb_nm_position(lyt, min_coord);
+        const auto view_box_nm_max = sidb_nm_position(lyt, shifted_max);
+
+        const auto viewbox_x      = view_box_nm_min.first * 10;
+        const auto viewbox_y      = view_box_nm_min.second * 10;
+        const auto viewbox_width  = (view_box_nm_max.first - view_box_nm_min.first) * 10;
+        const auto viewbox_height = (view_box_nm_max.second - view_box_nm_min.second) * 10;
+
+        // Generate background rectangle
+        const auto background_rect =
+            fmt::format(R"(<rect x="{0}" y="{1}" width="{2}" height="{3}" style="fill:{4};"/>)", viewbox_x, viewbox_y,
+                        viewbox_width, viewbox_height, background_color);
+
+        // Generate the final SVG content
+        os << fmt::format(fiction::svg::HEADER_TEMPLATE, FICTION_VERSION, FICTION_REPO, viewbox_x, viewbox_y,
+                          viewbox_width, viewbox_height, formatted_path_definition, background_rect, svg_content.str());
+    }
+};
+
+}  // namespace detail
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into an output stream.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param os The output stream to write into.
+ * @param ps Parameters.
+ *
+ * @note SiDB defects are not supported yet.
+ */
```

Comment:
Please move notes to directly after the explanation
```suggestion
/**
 * Writes an SVG representation of a cell-level SiDB layout into an output stream.
 *
 * @note SiDB defects are not supported yet.
 *
 * @tparam Lyt SiDB cell-level layout type.
 * @param lyt The layout to be written.
 * @param os The output stream to write into.
 * @param ps Parameters.
 */
```